catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/UpdateOperations.java

```diff
@@ -733,4 +736,37 @@ private UpdateStorageRequest applyAttributeOverrides(UpdateStorageRequest update
                 metacardMap);
         return updateStorageRequest;
     }
+
+    private UpdateRequest populateResourceUris(UpdateRequest updateRequest, QueryResponse query) {
+        Set<Metacard> updateRequestMetacards = updateRequest.getUpdates()
+                .stream()
+                .map(Map.Entry::getValue)
+                .filter(metacard -> metacard.getResourceURI() == null)
+                .collect(Collectors.toSet());
+
+        Map<String, Metacard> queryResultMetacards = query.getResults()
+                .stream()
+                .map(Result::getMetacard)
+                .filter(Objects::nonNull)
+                .filter(metacard -> metacard.getResourceURI() != null)
+                .collect(Collectors.toMap(Metacard::getId, Function.identity()));
+
+        for (Metacard updateMetacard : updateRequestMetacards) {
+            Metacard queriedMetacard = queryResultMetacards.get(updateMetacard.getId());
+            if (queriedMetacard != null && queriedMetacard.getId()
+                    .equals(updateMetacard.getId())) {
+                LOGGER.info(
+                        "Found URI \"{}\" missing from updated Metacard with ID \"{}\". Setting Metacard field \"{}\" with the value.",
```

Comment:
The specificity of this workaround concerns me. We have had multiple problems in the past with reconstructed metacards accidentally dropping attributes. I would prefer a more general solution, shouldn't we be checking the metacard's type to make sure we are not dropping anything?